aws/resource_aws_lambda_permission.go

```diff
@@ -225,6 +225,16 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		if err == nil {
 			var psErr error
 			statement, psErr = getLambdaPolicyStatement(out, d.Id())
+
+			// handle the resource not existing
+			if awsErr, ok := psErr.(awserr.Error); ok {
```

Comment:
Pattern taken from existing code a few lines below.  I don't believe we were ever hitting that case, but left just to be sure.